src/core/grid_based_algorithms/electrokinetics_cuda.cu

```diff
@@ -3641,100 +3642,143 @@ void ek_print_lbpar() {
+    throw std::runtime_error(
+        "Electrokinetics parameters cannot be set after initialisation");
   ek_parameters.viscosity = viscosity;
   return 0;
 }
 
 int ek_set_lb_ext_force_density(float lb_ext_force_dens_x,
                                 float lb_ext_force_dens_y,
                                 float lb_ext_force_dens_z) {
+  if (ek_initialized)
+    throw std::runtime_error(
+        "Electrokinetics parameters cannot be set after initialisation");
   ek_parameters.lb_ext_force_density[0] = lb_ext_force_dens_x;
   ek_parameters.lb_ext_force_density[1] = lb_ext_force_dens_y;
   ek_parameters.lb_ext_force_density[2] = lb_ext_force_dens_z;
   return 0;
 }
 
 int ek_set_friction(float friction) {
-
+  if (ek_initialized)
+    throw std::runtime_error(
+        "Electrokinetics parameters cannot be set after initialisation");
   ek_parameters.friction = friction;
   return 0;
 }
 
 int ek_set_bulk_viscosity(float bulk_viscosity) {
-
+  if (ek_initialized)
+    throw std::runtime_error(
+        "Electrokinetics parameters cannot be set after initialisation");
   ek_parameters.bulk_viscosity = bulk_viscosity;
   return 0;
 }
 
 int ek_set_gamma_odd(float gamma_odd) {
-
+  if (ek_initialized)
+    throw std::runtime_error(
+        "Electrokinetics parameters cannot be set after initialisation");
   ek_parameters.gamma_odd = gamma_odd;
   return 0;
 }
 
 int ek_set_gamma_even(float gamma_even) {
-
+  if (ek_initialized)
+    throw std::runtime_error(
+        "Electrokinetics parameters cannot be set after initialisation");
   ek_parameters.gamma_even = gamma_even;
   return 0;
 }
 
 int ek_set_stencil(int stencil) {
+  if (ek_initialized)
+    throw std::runtime_error(
+        "Electrokinetics parameters cannot be set after initialisation");
   if (!ek_parameters.fluidcoupling_ideal_contribution)
     return 1; // combination not implemented
-
   ek_parameters.stencil = stencil;
   return 0;
 }
 
 int ek_set_advection(bool advection) {
+  if (ek_initialized)
+    throw std::runtime_error(
+        "Electrokinetics parameters cannot be set after initialisation");
   ek_parameters.advection = advection;
   return 0;
 }
 
 int ek_set_fluctuations(bool fluctuations) {
+  if (ek_initialized)
+    throw std::runtime_error(
+        "Electrokinetics parameters cannot be set after initialisation");
   ek_parameters.fluctuations = fluctuations;
   return 0;
 }
 
 int ek_set_fluctuation_amplitude(float fluctuation_amplitude) {
+  if (ek_initialized)
+    throw std::runtime_error(
+        "Electrokinetics parameters cannot be set after initialisation");
   ek_parameters.fluctuation_amplitude = fluctuation_amplitude;
   return 0;
 }
 
 int ek_set_fluidcoupling(bool ideal_contribution) {
+  if (ek_initialized)
+    throw std::runtime_error(
+        "Electrokinetics parameters cannot be set after initialisation");
   if (ek_parameters.stencil != 0)
-    return 1; // combination not implemented
-
+    return 1; // combination not implemente
```

Comment:
The error code returned here is not handled in Cython. Please convert it to a `throw std::runtime_error()`. You can then convert all these setter functions to return type `void` and remove their `return 0;` (details in #4219).